src/core/src/MatrixDynSize.cpp

```diff
@@ -76,37 +77,32 @@ MatrixDynSize& MatrixDynSize::operator=(const MatrixDynSize& other)
 {
     if (this == &other) return *this;
 
-    // Copy the new rows
+    // Copy the new rows and columns
     m_rows = other.m_rows;
     m_cols = other.m_cols;
+    unsigned requiredCapacity = m_rows * m_cols;
+
+    // if other is empty, return
+    //FIXME: This probably changed the behaviour.
+    // the capacity is never changed if the other matrix is zero
+    if (requiredCapacity == 0) return  *this;
```

Comment:
I understand this is necessary to avoid passing a null pointer to 
~~~
std::memcpy(this->m_data, other.m_data, requiredCapacity * sizeof(double));
~~~
but I think it would more readable if we just put an if check before that memcpy, to avoid having multiple return points. 
Furthermore, I do not understand the comment. Which behaviour changed?